File: Python/3-LSP/LSP-Solucao/Quadrado.py
from Paralelogramo import Paralelogramo

class Quadrado(Paralelogramo):
    def __init__(self, altura, largura):
        super().__init__(altura, largura)
        
        if self._largura != self._altura:
            raise Exception('Os dois lados do quadrado precisam ser iguais')
        

File: PHP/3-LSP/LSP-Solucao/Quadrado.php
<?php
require_once('Paralelogramo.php');

class Quadrado extends Paralelogramo
{
    public function __construct($altura, $largura)
    {
        parent::__construct($altura, $largura);

        if ($this->largura != $this->altura) {
            throw new Exception("Os dois lados do quadrado precisam ser iguais");
        }
    } 
}
